### PR TITLE
395 time series rdb connection

### PR DESCRIPTION
**Pull request by kokfoongcmcl**
Externally, no change is required from codes using the TimeSeriesClient, all changes are internal.

Key changes:
1) RemoteRDBStoreClient no longer stores connection object, though at the moment there's no way of getting the privately stored "conn" variable, it's still best not to store it at all
2) Un-deprecated old time series methods without the connection argument, they all now work by opening a single connection using try-with-resource
3) Methods without the connection argument call the methods with the connection argument, thus removing code duplication
4) Renamed tests from deprecated to "WithoutConn"
5) duplicate tests are removed

### CODE REVIEW

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java [hunk 1]
```diff
@@ -107,17 +96,13 @@ public ResultSet executeQuerytoResultSet(String query) {
...
     @Override
     public JSONArray executeQuery(String query){
-        JSONArray results;
-        try {
-            if(this.conn == null || stmt==null){
-                this.conn = getConnection();
-            }
-            java.sql.ResultSet rs = this.stmt.executeQuery(query);
-            results = StoreClientHelper.convert(rs);
+        try (Connection conn = DriverManager.getConnection(this.rdbURL, this.rdbUser, this.rdbPassword);
```
Comment (gpeb2) [inline JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java:99]: Same as above.

**Review by gpeb2 — CHANGES_REQUESTED** [review]

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java [hunk 2]
```diff
@@ -47,6 +48,8 @@
 	private TimeSeriesSparql rdfClient;
 	// Exception prefix
 	private final String exceptionPrefix = this.getClass().getSimpleName() + ": ";
+	private static final String CONNECTION_ERROR = "Failed to connect to database. If you are using the methods without the connection argument, " +
+	"the RDB endpoint (URL, username and password) needs to be set in the constructor of TimeSeriesClient";
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java:56]: yes

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java [hunk 3]
```diff
@@ -259,32 +289,34 @@ public void deleteTimeSeries(String tsIRI, Connection conn) {
...
 			// try to revert previous knowledge base deletion
 			// TODO Ideally try to avoid throwing exceptions in a catch block - potential solution: have initTS throw
 			//		a different exception depending on what the problem was, and how it should be handled
+
+			// modification mainly to make tests work when the Connection is a MockConnection object, rdbUrl is not used anyway
+			String rdbUrl;
+			try {
+				rdbUrl = conn.getMetaData().getURL();
+			} catch (Exception e) {
+				rdbUrl = ""; // setting dummy url, is not used in practice anyway
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/77b24f93b2c69ba21818d9165c52572d83e56afb and https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/4cc5d30d3f2b167eae0da1c5580100082841d464

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java [hunk 4]
```diff
@@ -83,38 +90,61 @@ public void initTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, Stri
     	// Step1: Initialise time series in knowledge base
     	// In case any exception occurs, nothing will be created in kb, since JPSRuntimeException will be thrown before 
     	// interacting with triple store and SPARQL query is either executed fully or not at all (no partial execution possible)
+		
+		// modification to make tests work when the Connection is a MockConnection object, rdbUrl is not used anyway 
+		String rdbURL;
+		try {
+			rdbURL = conn.getMetaData().getURL();		
+		} catch (Exception e) {
+			rdbURL = "";
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java:109]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/77b24f93b2c69ba21818d9165c52572d83e56afb and https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/4cc5d30d3f2b167eae0da1c5580100082841d464

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java [hunk 5]
```diff
@@ -107,17 +96,13 @@ public ResultSet executeQuerytoResultSet(String query) {
...
-        JSONArray results;
-        try {
-            if(this.conn == null || stmt==null){
-                this.conn = getConnection();
-            }
-            java.sql.ResultSet rs = this.stmt.executeQuery(query);
-            results = StoreClientHelper.convert(rs);
+        try (Connection conn = DriverManager.getConnection(this.rdbURL, this.rdbUser, this.rdbPassword); 
+            Statement stmt = conn.createStatement(java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY)) {
+                java.sql.ResultSet rs = stmt.executeQuery(query);
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/379c623838c443e0a9965d65192f07a64edab4e4

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java [hunk 6]
```diff
@@ -87,16 +80,12 @@ public Connection getConnection(){
      * @return query result as a ResultSet
      */
     public ResultSet executeQuerytoResultSet(String query) {
-        ResultSet rs;
-        try {
-            if(this.conn == null || stmt==null){
-                this.conn = getConnection();
-            }
-            rs = this.stmt.executeQuery(query);
+        try (Connection conn = DriverManager.getConnection(this.rdbURL, this.rdbUser, this.rdbPassword);
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/20a7ade613b1bd16ea00aa46a54f960fceeb138b

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java (hunk 1 above)
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/20a7ade613b1bd16ea00aa46a54f960fceeb138b

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java [hunk 7]
```diff
@@ -160,13 +145,11 @@ public String execute(String query){
      */
     @Override
     public int executeUpdate(String update) {
-        try {
-            if(this.conn == null || stmt==null){
-                this.conn = getConnection();
-            }
-            return this.stmt.executeUpdate(update);
+        try (Connection conn = DriverManager.getConnection(this.rdbURL, this.rdbUser, this.rdbPassword);
```
[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java]: Addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/397/commits/20a7ade613b1bd16ea00aa46a54f960fceeb138b

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (gpeb2) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java:109]: Ideally the MockConnection class would be overridden to provide the required methods,

File: JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gpeb2) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java]: See other comment https://github.com/cambridge-cares/TheWorldAvatar/pull/397#discussion_r1009750910

[the remark this answers is not in the document]
Comment (kokfoongcmcl) [reply JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java]: addressed in latest commits

**Review by gpeb2 — APPROVED** [review]

**Pull request merged**